### PR TITLE
Dedicated admin can't manage upgradeconfig test.

### PR DESCRIPTION
**Pull request by aliceh**
In this commit, I added a test to verify that dedicated admin can't manage upgradeconfig and cluster admin can.

### CODE REVIEW

File: pkg/e2e/operators/managedupgrade.go [hunk 1]
```diff
@@ -177,6 +179,20 @@ var _ = ginkgo.Describe(managedUpgradeOperatorTestName, func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		ginkgo.It("dedicated admin should not be able to manage the UpgradeConfig CR", func() {
+			// Add the upgradeconfig to the cluster
+			uc := makeMinimalUpgradeConfig(upgradeConfigResourceName, operatorNamespace)
+			err = dedicatedAaddUpgradeConfig(uc, operatorNamespace, h)
+			Expect(apierrors.IsForbidden(err)).To(BeTrue())
+
+			err := addUpgradeConfig(uc, operatorNamespace, h)
```
Comment (cblecker) [inline pkg/e2e/operators/managedupgrade.go:188]: should we clean up this object when we are done with this test?

File: pkg/e2e/operators/managedupgrade.go [hunk 2]
```diff
@@ -30,7 +32,7 @@ var _ = ginkgo.Describe(managedUpgradeOperatorTestName, func() {
 	var operatorName = "managed-upgrade-operator"
 	var operatorNamespace string = "openshift-managed-upgrade-operator"
 	var operatorLockFile string = "managed-upgrade-operator-lock"
-	var upgradeConfigResourceName string = "osd-upgrade-config"
+	var upgradeConfigResourceName string = "osde2e-upgrade-config"
```
Comment (cblecker) [inline pkg/e2e/operators/managedupgrade.go:35]: Does changing this alter any other test?

**Comment by aliceh** [comment]
@cblecker, thanks for reviewing and the suggestions! I updated the code accordingly.

**Pull request merged**